polygon/sync/block_downloader.go

```diff
@@ -196,6 +206,9 @@ func (d *blockDownloader) downloadBlocksUsingWaypoints(ctx context.Context, wayp
 					return
 				}
 
+				blocksTotalSize += totalSize
```

Comment:
`blocksTotalSize` and `blockCount` need to be `atomic.Int64` and use `.Add` here to avoid race conditions (these increments are done in goroutines in parallel)